File: Problem 9 - Palindrome Number/code.js
var isPalindrome = function(x) {
    if (x < 0 || (x % 10 === 0 && x !== 0)) {   
        return false;
    }

    let reversed = 0;
    let original = x;

    while (original > reversed) {
        reversed = reversed * 10 + original % 10;
        original = Math.floor(original / 10);
    }

    return original === reversed || original === Math.floor(reversed / 10);
}

// Case 1 
// Input: x = 121
// Expected Output: true
console.log(isPalindrome(121))

// Case 2
// Input: x = -121
// Expected Output: false
console.log(isPalindrome(-121))

// Case 3
// Input: x = 10
// Expected Output: false
console.log(isPalindrome(10))

File: Problem 9 - Palindrome Number/code.py
import math

class Solution(object):
    def isPalindrome(self, x):
        if x < 0 or x % 10 == 0 and x != 0:
            return False

        reversed = 0
        original = x

        while original > reversed:
            reversed = reversed * 10 + original % 10
            original = math.floor(original / 10)

        return original == reversed or original == math.floor(reversed / 10)
    
    # Case 1 
    # Input: x = 121
    # Expected Output: True
    print(isPalindrome("", 121))

    # Case 2
    # Input: x = -121
    # Expected Output: False
    print(isPalindrome("", -121))

    # Case 3
    # Input: x = 10
    # Expected Output: False
    print(isPalindrome("", 10))
        